### DISCUSSION TITLE
Build Instructions

### DISCUSSION THREAD

**Discussion by PaulDuncan-SOI** [jaeaton/ECWP-Winch-Data#2] [General]
Hi,

I thought I would kick things off. Downloaded the source, and it is not a language I am familiar with (I've done Unix/C and a little Python 3 in the past). I think it may be a special version of .Net. So, questions (which maybe others will find useful):

   Can it only be built on WIndows, and you select the targets that you need to build for? Or are there build tools for this on Linux?

   If it can be worked on on a Linux platform, what tools should be used?

Thanks!

Paul.

**Comment by jaeaton** [comment]
Paul,

I do my development in Windows with Visual Studio. It is written in C# with a frontend of Avalonia. C# is very C/C++ like. Avalonia is similar to WPF. I think you can choose to use the .net SDK 6.0 or the mono SDK. https://learn.microsoft.com/en-us/dotnet/core/install/linux

I think the easiest way to start would be to use Visual Studio Code with the C# extension installed. https://dotnet.microsoft.com/en-us/platform/tools

It has been a long time since I have run a Linux Box. I have a virtual Linux machine installed and have done some testing with that, but wasn't able to test everything due to the quirks of virtualization.

Josh

From: SOI-Jimbo ***@***.***>
Sent: Saturday, February 18, 2023 2:47 PM
To: jaeaton/ECWP-Winch-Data ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [jaeaton/ECWP-Winch-Data] Build Instructions (Discussion #2)


Hi,

I thought I would kick things off. Downloaded the source, and it is not a language I am familiar with (I've done Unix/C and a little Python 3 in the past). I think it may be a special version of .Net. So, questions (which maybe others will find useful):

Can it only be built on WIndows, and you select the targets that you need to build for? Or are there build tools for this on Linux?

If it can be worked on on a Linux platform, what tools should be used?

Thanks!

Paul.

-
